clang/lib/Sema/SemaDecl.cpp

```diff
@@ -3217,6 +3217,22 @@ static void adjustDeclContextForDeclaratorDecl(DeclaratorDecl *NewD,
     FixSemaDC(VD->getDescribedVarTemplate());
 }
 
+template <typename AttributeName>
+static void checkDimensionsAndSetDiagnostics(Sema &S, FunctionDecl *New,
+                                             FunctionDecl *Old) {
+  auto *NewDeclAttr = New->getAttr<AttributeName>();
+  auto *OldDeclAttr = Old->getAttr<AttributeName>();
+  if ((NewDeclAttr->getXDim() != OldDeclAttr->getXDim()) ||
+      (NewDeclAttr->getYDim() != OldDeclAttr->getYDim()) ||
+      (NewDeclAttr->getZDim() != OldDeclAttr->getZDim())) {
+    S.Diag(New->getLocation(), diag::err_conflicting_sycl_function_attributes)
+        << OldDeclAttr << NewDeclAttr;
+    S.Diag(New->getLocation(), diag::warn_duplicate_attribute) << OldDeclAttr;
+    S.Diag(OldDeclAttr->getLocation(), diag::note_conflicting_attribute);
```

Comment:
I think you can use location of the functions instead of locations of the attributes, right?

Reply:
I used location of attributes to set the diagnostic on that specific line instead of the function name line. This seems to be the practice in other places too, e.g., MarkDevices function in SemaSYCL.cpp

Reply:
Just curious then, why the first two diagnostic emissions use location of the function?

Reply:
I am using the location of attributes only to produce the note about where the conflicting attributes exist but using function location for error or warning.